### PR TITLE
DAV SEARCH: Expose upload_time and creation_time

### LINKED ISSUE #21385: Upload time and Creation time in searchrequest
https://github.com/nextcloud/server/pull/17765

```
<?xml version="1.0"?>
<d:searchrequest xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"> 
<d:basicsearch>
...
<d:lte>
  <d:prop>
    <nc:upload_time />
  </d:prop>
  <d:literal>1591716364</d:literal>
</d:lte>
...
```

return error, isn't  implemented?

### PR DESCRIPTION
**Pull request by marcelklehr**
fixes #21385

This would be very useful for https://github.com/nextcloud/photos/issues/87

### CODE REVIEW

**Comment by marcelklehr** [comment]
php-cs linter fails due to unrelated file `lib/private/Files/ObjectStore/S3ConnectionTrait.php`

**Review by juliusknorr — APPROVED** [review]

**Comment by marcelklehr** [comment]
@icewind1991 Does Node#getFileInfo()->getCreationTime() return the file system creation time, or some value maintained in nextcloud? In my tests with this PR it always returns `0`.

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 1]
```diff
@@ -404,6 +404,9 @@ public function handleGetProperties(PropFind $propFind, \Sabre\DAV\INode $node)
 			$propFind->handle(self::CREATION_TIME_PROPERTYNAME, function () use ($node) {
 				return $node->getFileInfo()->getCreationTime();
 			});
+			$propFind->handle(self::CREATIONDATE_PROPERTYNAME, function () use ($node) {
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php:407]:
```suggestion
			$propFind->handle(self::CREATIONDATE_PROPERTYNAME, function () use ($node): int {
```

**Comment by CarlSchwan** [comment]
> Does Node#getFileInfo()->getCreationTime() return the file system creation time, or some value maintained in nextcloud? In my tests with this PR it always returns 0.

From that I understand the creation_time is set here: https://github.com/nextcloud/server/blob/8367b02785acd920db244ecb745517820ed9ff91/apps/dav/lib/Connector/Sabre/File.php#L379

So a client needs to send the time explicitly when uploading a file. Reading the desktop client source code, I can't find where this is set :(

**Comment by PVince81** [comment]
maybe @mgallien for where the creation time is sent ?

**Comment by szaimen** [comment]
Hi everyone, what is needed to get this done? :)

**Comment by marcelklehr** [comment]
This was based on wrong assumptions

**Pull request closed without merging**